pkg/hive/hive_test.go

```diff
@@ -380,3 +422,43 @@ func readAndAssertPeersMsgs(in []byte, expectedLen int) ([]pb.Peers, error) {
 
 	return peers, nil
 }
+
+func ComparePeerMsgs(got, want []*pb.BzzAddress) error {
+	gotMap := flattenByOverlay(got)
+	wantMap := flattenByOverlay(want)
+
+	for ovlHex, w := range wantMap {
+		g, ok := gotMap[ovlHex]
+		if !ok {
+			return fmt.Errorf("expected peer %s, but not found", ovlHex)
+		}
+		if !bytes.Equal(g.Underlay, w.Underlay) {
+			return fmt.Errorf("peer %s: underlays (got=%s want=%s)", ovlHex, g.Underlay, w.Underlay)
+		}
+		if !bytes.Equal(g.Signature, w.Signature) {
+			return fmt.Errorf("peer %s: expected signatures (got=%s want=%s)",
+				ovlHex, shortHex(g.Signature), shortHex(w.Signature))
+		}
+		if !bytes.Equal(g.Nonce, w.Nonce) {
+			return fmt.Errorf("peer %s: expected nonce (got=%s want=%s)",
+				ovlHex, shortHex(g.Nonce), shortHex(w.Nonce))
+		}
+	}
+	return nil
+}
+
+func flattenByOverlay(batches []*pb.BzzAddress) map[string]*pb.BzzAddress {
+	m := make(map[string]*pb.BzzAddress)
+	for _, batch := range batches {
+		m[hex.EncodeToString(batch.Overlay)] = batch
```

Comment:
This will replace any previously assigned value for this key. Is this the desired behaviour?